models/articles.models.js

```diff
@@ -0,0 +1,23 @@
+const db = require("../db/connection");
+
+exports.fetchArticles = () => {
+  return db
+    .query(
+      `SELECT * FROM articles LEFT OUTER JOIN comments ON comments.created_at = articles.created_at ORDER BY articles.created_at DESC;`
+    )
+    .then(({ rows }) => {
+      const addCommentCount = rows.map((eachObj) => {
```

Comment:
refactor to use an aggregate function, like `COUNT()`, to add comment count to your query. You can even use `AS` to put it on the key comment_count.

Reply:
Hi, Alex I tried the aggregate function `SELECT comments.*, COUNT(article_id) AS comment_count FROM comments JOIN articles ON articles.created_at  = comments.created_at GROUP BY article_id;`
It returns **GROUP BY article_id** is _ambiguous_ but at least JavaScript can read nearly the end of the code.

Then I tried`SELECT comments.*, COUNT(article_id) AS comment_count FROM comments JOIN articles ON articles.created_at  = comments.created_at GROUP BY **articles.article_id**;`. It just reveals the beginning of COUNT(article_id) is _ambiguou_.

I just used the method taught in the note, copied it exactly the same. It not the problem of join method, even I tried LEFT, RIGHT etc. I still cannot run the code!

Reply:
I tired `SELECT comments.*, COUNT(article_id) AS comment_count FROM comments JOIN articles ON articles.created_at  = comments.created_at GROUP BY comments.article_id;` too, It reveals the beginning of COUNT(article_id) is ambiguous again

Reply:
I also use 2 variable to contain 2 tables' value. I make the comments variable into (Key)the article_id and (Value)
When Im using map method in articles array of each object, I assign comment_count into it. But I can only get NAN.
At the same time, it is not an appropriate method to separate 2 tables and return into 2 arrays because JS shows cannot put 2 Pools(DB) together

Reply:
> article_id



> I tired `SELECT comments.*, COUNT(article_id) AS comment_count FROM comments JOIN articles ON articles.created_at = comments.created_at GROUP BY comments.article_id;` too, It reveals the beginning of COUNT(article_id) is ambiguous again

Hi Catherine, this is really close! If it's saying this is ambiguous it means something hasn't been specified that could mean multiple things - your article_id (in `COUNT(article_id)` may be ambiguous because article_id exists in multiple tables, how could you specify which one?

Reply:
We do have tickets in a specific order for a reason though Catherine, so please do continue with this ticket before moving onto the next one 🙏 you were really close!
You can do nchelp if needed 🙂